internal/services/magic_transit_connector/resource.go

```diff
@@ -112,20 +135,13 @@ func (r *MagicTransitConnectorResource) Update(ctx context.Context, req resource
 		return
 	}
 
-	dataBytes, err := data.MarshalJSONForUpdate(*state)
-	if err != nil {
-		resp.Diagnostics.AddError("failed to serialize http request", err.Error())
-		return
-	}
 	res := new(http.Response)
 	env := MagicTransitConnectorResultEnvelope{*data}
-	_, err = r.client.MagicTransit.Connectors.Update(
+
+	_, err := r.client.MagicTransit.Connectors.Edit(
```

Comment:
@vaishakdinesh The API to use needs to be `Edit`, not `Update`. (patch vs. put)

Reply:
It worked... sort of. The JSON patch update works. Although this is because of the side effect that `device` is marked as required and never change in a `Update` call, so it won't include the `device` when making the JSON edit params in this function.